src/lib.rs

```diff
@@ -1,14 +1,248 @@
+
+        let led = Led::new();
+
+        Self {
+            adc: (adc),
+            bmp: (bmp),
+            pwm: (pwm),
+            mag: (mag),
+            imu: (imu),
+            led: (led),
+        }
+    }
+
+    pub fn init(&mut self) {
+        self.self_test();
+        //Initialize devices on navigator's default settings,
+        //read more on ./navigator-api.pdf
+        self.mag.init().unwrap();
+        self.mag.set_mode(mag_Mode::Cont200Hz).unwrap();
+
+        self.imu
+            .setup(&mut Delay {})
+            .expect("Error: Failed on IMU setup");
+        self.imu.set_accel_range(AccelRange::Range_2g).unwrap();
+        self.imu.set_gyro_range(GyroRange::Range_250dps).unwrap();
 
-    #[test]
-    fn it_works() {
-        let result = add(2, 2);
-        assert_eq!(result, 4);
+        self.adc.reset_internal_driver_state();
+
+        self.pwm.reset_internal_driver_state();
+        self.pwm.use_external_clock().unwrap();
+        self.pwm.set_prescale(100).unwrap();
+        self.pwm.enable().unwrap();
+
+        self.bmp.zero().unwrap();
+
+        self.led.all_off()
+    }
+
+    pub fn self_test(&mut self) -> bool {
+        //Check if the sensors are attached by it's IDs,
+        //run self-test if they have.
+        self.imu.check_identity(&mut Delay {}).unwrap();
+        self.mag
+            .self_test()
+            .expect("Error : Error on magnetometer during self-test")
+    }
+
+    pub fn set_pwm_value(&mut self) {
+        let channel = pwm_Channel::C0; //todo import enums
+        let value = 2047; //todo a fomula that validades from 0 to 100.00% and convert to 4092
+        self.pwm.set_channel_on(channel, 0).unwrap(); //Channel::C0
+        self.pwm.set_channel_off(channel, value).unwrap();
+    }
+
+    pub fn set_pwm_freq(&mut self) {
+        todo!()
+    }
+
+    pub fn set_pwm_off(&mut self) {
+        todo!()
+    }
+
+    pub fn set_pwm_on(&mut self) {
+        todo!()
+    }
+
+    pub fn set_led_on(&mut self) {
+        self.led.all_on()
+    }
+    pub fn set_led_off(&mut self) {
+        self.led.all_off()
+    }
+
+    pub fn read_mag(&mut self) -> [f32; 3] {
+        let (x, y, z) = self.mag.read().unwrap();
+        [x.into(), y.into(), z.into()]
+    }
+
+    pub fn read_bmp(&mut self) -> [f32; 3] {
+        let pressure = self.bmp.pressure_kpa().unwrap();
+        let altitude = self.bmp.altitude_m().unwrap();
+        let temperature = self.bmp.temperature_celsius().unwrap();
+        [altitude, pressure, temperature] //fix here struc.
+    }
+
+    pub fn read_temperature(&mut self) -> f32 {
+        //this come from bmp device
+        self.bmp.temperature_celsius().unwrap()
+    }
+
+    pub fn read_altitude(&mut self) -> f32 {
+        //this come from bmp device
+        self.bmp.altitude_m().unwrap()
+    }
+    pub fn read_pressure(&mut self) -> f32 {
+        //this come from bmp device
+        self.bmp.pressure_kpa().unwrap()
```

Comment:
The code is pretty clear about what sensor gives pressure, you can remove these comments